ament_package/url.py

```diff
@@ -16,9 +16,9 @@
 class Url(object):
     __slots__ = ['url', 'type']
 
-    def __init__(self, url, type_=None):
+    def __init__(self, url, *, type=None):
```

Comment:
I think `type_` is better than `type`, because you could still use the builtin `type` function in the constructor. But I agree that neither are good names because of the symbol collision.

You could use `url_type` or `site_type`?